flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -1520,6 +1523,44 @@ private void cancelAllNotifications(Result result) {
     result.success(null);
   }
 
+  public void requestPermission(@NonNull PermissionRequestListener callback) {
+    if (permissionRequestInProgress) {
+      callback.complete(false);
```

Comment:
Looks like this would complete and indicate to users of the API that the permission wasn't granted. This could be misleading as apps might think the user denied the permissions and the app updates the UI to present some text around this. IMO it would be better to return an error so apps can choose treat this differently e.g. swallow the error. The approach you're using here is similar to https://github.com/Baseflow/flutter-permission-handler/blob/051828f0c5b572fde65ef3474af0de29407fe9aa/permission_handler_android/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java#L210 so triggering an error callback could be done to solve this

Reply:
You're right, thank you. I fixed this.